app/controllers/companies_controller.rb

```diff
@@ -4,7 +4,7 @@
 
 class CompaniesController < ApplicationController
   def index
-    @companies = CompaniesSelector.new(params[:query]).select
+    @companies = CompaniesSelector.new(params).select
```

Comment:
I think it is not a good idea to pass all the params (user could add unlimited nomber of params) to the service. I think it is better to filter them out.

```ruby
@companies = CompaniesSelector.new(stock: params[:stock], query: params[:query]).select
```

or

```ruby
@companies = CompaniesSelector.new(params.select(:stock, :query]).select
```

Reply:
I think this may work:

```ruby
@companies = CompaniesSelector.new(params.slice(:stock, :query)).select
```

Reply:
Actually, you probably should use [ActionController::Parameters](https://api.rubyonrails.org/classes/ActionController/Parameters.html):

```ruby
@companies = CompaniesSelector.new(params.permit(:stock, :query)).select
end

Reply:
Ok, good idea, so as I understand the `params` which is passed by request is already object of Parameter class?

Reply:
Probably yes, I think Rails provides you params as an instance of this class.

You can use debugger and check it by typing `params.class`.